plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -1131,6 +1131,28 @@ public void testIoExplain()
         assertUpdate("DROP TABLE test_io_explain");
     }
 
+    @Test
+    public void testIoExplainWithEmptyPartitionedTable()
+    {
+        // Test IO explain a partitioned table with no data.
+        computeActual("CREATE TABLE test_io_explain WITH (partitioned_by = ARRAY['orderkey']) AS SELECT custkey, orderkey FROM orders WITH NO DATA");
+
+        EstimatedStatsAndCost estimate = new EstimatedStatsAndCost(0.0, 0.0, 0.0, 0.0, 0.0);
+        MaterializedResult result = computeActual("EXPLAIN (TYPE IO, FORMAT JSON) SELECT custkey, orderkey FROM test_io_explain");
+        assertEquals(
+                getIoPlanCodec().fromJson((String) getOnlyElement(result.getOnlyColumnAsSet())),
+                new IoPlan(
+                        ImmutableSet.of(
+                                new TableColumnInfo(
+                                        new CatalogSchemaTableName(catalog, "tpch", "test_io_explain"),
+                                        ImmutableSet.of(),
+                                        estimate)),
+                        Optional.empty(),
+                        estimate));
+
+        assertUpdate("DROP TABLE test_io_explain");
```

Comment:
test_io_explain -> test_io_explain_with_empty_partitioned_table

using same table name as in the other test will make tests fail when run concurrently

Reply:
https://github.com/trinodb/trino/pull/12349#discussion_r876894886
+ IoPlanPrinter.parseConstraints will check the constraint is none or not. The logic is correct.
+ I think if a partitioned table do not have any data, we should return TupleDomain.all() as constraint. there is only one case that partitions is empty and partitionColumns is not because of no data. 
+ Only partitionColumns and partitions are empty, we should return TupleDomain.none().
+ What is your thought?

Reply:
+ This implement will not take effect  "no rows match the filter". because there is no data(partitions is empty and partitionColumns is not empty), even we return TupleDomain.all(), there is no data match the filter, the result is same.
+ I agree with you, this is better to fix inside IoPlanPrinter.parseConstraints, but i still do not know how to get partitions informations in plan process. i will continue to think about it tomorrow. if you have free time, please help to give me some suggestion. 

Thanks for your time
Alberic

Reply:
> even we return TupleDomain.all(), there is no data match the filter, the result is same.

`io.trino.spi.connector.ConnectorMetadata#getTableProperties`'s `io.trino.spi.connector.ConnectorTableProperties#predicate` is used to inform the planner and allow deriving filters for other tables.

for example, a query
```sql
SELECT * FROM some_table JOIN empty_partitioned_table ON ...
```

should be reduced to `SELECT .. WHERE false` because the planner realizes `empty_partitioned_table` has no rows (`TupleDomain.none()`).